Codeinwp/optimole-wp
# Dashboard to React

### All Submissions:

 

### Changes proposed in this Pull Request:

This PR moves Optimole Dashboard from Vue JS to React using wp-scripts & WordPress components.

- All used parts from Vue have been moved to React.
- Unused dependencies have been removed, and build process has been moved to wp-scripts
- Using Tailwind for CSS.

**Important:** One thing to confirm is that it will require a minimum version to 6.0. All things work below 6.0 apart from the Redux store. It might require some work to migrate the store to work on older versions. createReduxStore version on older versions doesn't support how we use dispatch inside actions.
 
<img width="2395" alt="Screenshot 2023-06-08 at 7 35 13 PM" src="https://github.com/Codeinwp/optimole-wp/assets/2649903/91fc7eb3-82cd-43a4-a4c2-13ef0dafaad4">
<img width="2396" alt="Screenshot 2023-06-08 at 7 35 34 PM" src="https://github.com/Codeinwp/optimole-wp/assets/2649903/70f72b44-ee40-49e6-8509-fb8a1b9356f2">
<img width="2396" alt="Screenshot 2023-06-08 at 7 35 45 PM" src="https://github.com/Codeinwp/optimole-wp/assets/2649903/bf643f0a-447b-4a5d-883a-b55b46c3f8d6">
<img width="2397" alt="Screenshot 2023-06-08 at 7 36 08 PM" src="https://github.com/Codeinwp/optimole-wp/assets/2649903/c7447846-a7cf-48dd-8488-127e9ff5cffa">

 

Closes [#816](https://github.com/Codeinwp/optimole-service/issues/816).

### How to test the changes in this Pull Request:

1. Make sure everything works the same as it did with older Dashboard.
2. All the options are in the same place, so just make sure they work the same way.
3. It will also be good to switch settings in one dashboard and installing the other to make sure the same changes show up in both versions.

### Other information:
